docs/authoring.md

```diff
@@ -0,0 +1,93 @@
+# Authoring Components
+
+## Overview
+C#/WinRT provides support for authoring Windows Runtime components. You can write a library in C#, and use C#/WinRT's source generator to get a winmd that any WinRT compatible language can use. For example, a library written in C# can be used by a C++ program, via C#/WinRT and C++/WinRT, with just a few tweaks to the C++ project.
+
+
+## Authoring the C# Component
+To create a library, select the Class Library (.NET Core) template in Visual Studio. 
+Make sure the `<TargetFramework>` for the project is one of `net5.0-windows10.0.19041.0`, `net5.0-windows10.0.18362.0` or `net5.0-windows10.0.17763.0`. 
+And add the following to your library's project file:
+```
+  <PropertyGroup>
+    <CsWinRTWindowsMetadata>10.0.19041.0</CsWinRTWindowsMetadata>
+    <CsWinRTComponent>true</CsWinRTComponent>
+    <CsWinRTEnableLogging>true</CsWinRTEnableLogging>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <GeneratedFilesDir Condition="'$(GeneratedFilesDir)'==''">$([MSBuild]::NormalizeDirectory('$(MSBuildProjectDirectory)', '$(IntermediateOutputPath)', 'Generated Files'))</GeneratedFilesDir>
+  </PropertyGroup>
+```
+And don't forget to include a `PackageReference` to `Microsoft.Windows.CsWinRT`!
+
+
+## Known Authoring Issues
+1. There are some programs you could write as/in your component that aren't available in the Windows Runtime. 
+We are working on implementing diagnostics in our tool that will catch these errors before a winmd is generated for your component.
+
+2. Not all C# types have been mapped to runtime types, but we are working on completing this coverage. 
+
+3. Composable type support is still in progress
+
+## Using the authored component in C++

```

Comment:
Is using the authored component in a C# app also possible or is that not relevant? and would there be different modifications for that scenario or just adding a reference to the winmd?

Reply:
I think it is possible; I believe that is what @AdamBraden is doing [here](https://github.com/AdamBraden/MyRandom/blob/64676c3b9f2a16fde36c158904e343626814e284/CsTestApp/CsTestApp.csproj#L9)

Which makes me think the consuming C# app just needs a project reference to the authored component? Or perhaps a package reference if they aren't to live in the same directory/tree

Reply:
C# apps can consume it and it is just a regular project/package reference like they would for any other .NET library.